### ISSUE TITLE
BUG in AdversarialFairness

### ISSUE THREAD

**Issue by SeanMcCarren** [fairlearn/fairlearn#1209]
#### Describe the bug
When automatically inferring categorical targets, we chose the wrong loss function. 

The code change to fix it is simple, in `fairlearn.adversarial._pytorch_engine.py:145`, use `torch.nn.CrossEntropyLoss` instead of `torch.nn.NLLLoss`. However, I want to discuss if it is worth it to fix it, or we should improve this in general.

#### Steps/Code to Reproduce
```
import pandas as pd
from sklearn.model_selection import train_test_split
from fairlearn.adversarial._adversarial_mitigation import AdversarialFairnessClassifier
from torch.nn import Softmax
d = {
     'col1': [1,2,3,4,5,6,7,8],
     'sex': [1,1,0,0,1,0,1,1],
     'col2': [11, 12,13,14,11, 12,13,14],
     'Target':[0,1,2,3,2,1,3,0]
     }
df = pd.DataFrame(data=d)

X = df[['col1','sex','col2']]
y = df[['Target']]
z = df["sex"] # In this example, we consider 'sex' the sensitive feature.
mitigator = AdversarialFairnessClassifier(
    backend='torch')
X_train, X_test, Y_train, Y_test, Z_train, Z_test = train_test_split(
    X, y, z, test_size=0.1
)

mitigator.fit(X_train, Y_train, sensitive_features=Z_train)
```

**Comment by hildeweerts** [comment]
Thanks, @SeanMcCarren! 

I would suggest fixing the bug and opening another issue to discuss more general/extensive improvements.

**Comment by Tizpaz** [comment]
Thanks @SeanMcCarren and @hildeweerts. I can start looking into this and fixing the bug.

**Comment by hildeweerts** [comment]
Amazing @Tizpaz! I have assigned the issue to you. Let me know if you have any questions!

**Comment by hildeweerts** [comment]
Closed by #1226
